server/src/server.ts

```diff
@@ -98,6 +99,20 @@ function getDocumentSettings(resource: string): Thenable<ICheckStyleSettings> {
     return result;
 }
 
+async function ensureConfigurationFileParam(config: string): Promise<string> {
+    switch (config.toLowerCase()) {
+        case 'google_checks':
+        case 'sun_checks':
+            return `/${config.toLowerCase()}.xml`;
+        default:
+            if (await pathExists(config)) {
+                return config;
+            } else {
+                throw new Error(`The configuration file ${config} does not exist`);
+            }
+    }
+}
+
 documents.onDidClose((event: TextDocumentChangeEvent) => documentSettings.delete(event.document.uri));
 documents.onDidOpen((event: TextDocumentChangeEvent) => checkstyle(event.document.uri));
 documents.onDidSave((event: TextDocumentChangeEvent) => checkstyle(event.document.uri));
```

Comment:
what will happen if checkstyle method throws exception. Is it handled by vscode?

Reply:
https://github.com/jdneo/vscode-checkstyle/blob/389f830d499fa0d0875746694451c8cb2eaf6335/server/src/server.ts#L143

there is a try catch block in this method

Reply:
Sorry I misunderstood. I should move the method which might throw error into the try-catch block, like the ensureConfigurationFileParam().